ports/re2/common.mk

```diff
@@ -0,0 +1,97 @@
+ifndef QCONFIG
+QCONFIG=qconfig.mk
+endif
+include $(QCONFIG)
+
+include $(MKFILES_ROOT)/qmacros.mk
+
+# Prevent qtargets.mk from re-including qmacros.mk
+define VARIANT_TAG
+endef
+
+NAME=re2
+
+DIST_BASE=$(PRODUCT_ROOT)/../../../re2
+
+ifdef QNX_PROJECT_ROOT
+DIST_BASE=$(QNX_PROJECT_ROOT)
+endif
+
+#$(INSTALL_ROOT_$(OS)) is pointing to $QNX_TARGET
+#by default, unless it was manually re-routed to
+#a staging area by setting both INSTALL_ROOT_nto
+#and USE_INSTALL_ROOT
+INSTALL_ROOT ?= $(INSTALL_ROOT_$(OS))
+
+#A prefix path to use **on the target**. This is
+#different from INSTALL_ROOT, which refers to a
+#installation destination **on the host machine**.
+#This prefix path may be exposed to the source code,
+#the linker, or package discovery config files (.pc,
+#CMake config modules, etc.). Default is /usr/local
+PREFIX ?= /usr/local
+
+#choose Release or Debug
+CMAKE_BUILD_TYPE ?= Release
+
+#override 'all' target to bypass the default QNX build system
+ALL_DEPENDENCIES = re2_all
+.PHONY: re2_all install check clean
+
+FLAGS   += -g -D_QNX_SOURCE
+LDFLAGS += -Wl,--build-id=md5
+
+include $(MKFILES_ROOT)/qtargets.mk
+
+#Search paths for all of CMake's find_* functions --
+#headers, libraries, etc.
+#
+#$(QNX_TARGET): for architecture-agnostic files shipped with SDP (e.g. headers)
+#$(QNX_TARGET)/$(CPUVARDIR): for architecture-specific files in SDP
+#$(INSTALL_ROOT)/$(CPUVARDIR): any packages that may have been installed in the staging area
+CMAKE_FIND_ROOT_PATH := $(QNX_TARGET);$(QNX_TARGET)/$(CPUVARDIR);$(INSTALL_ROOT)/$(CPUVARDIR)
+
+#Path to CMake modules; These are CMake files installed by other packages
+#for downstreams to discover them automatically. We support discovering
+#CMake-based packages from inside SDP or in the staging area.
+#Note that CMake modules can automatically detect the prefix they are
+#installed in.
+CMAKE_MODULE_PATH := $(QNX_TARGET)/$(CPUVARDIR)/$(PREFIX)/lib/cmake;$(INSTALL_ROOT)/$(CPUVARDIR)/$(PREFIX)/lib/cmake
+
+#Headers from INSTALL_ROOT need to be made available by default
+#because CMake and pkg-config do not necessary add it automatically
+#if the include path is "default"
+CFLAGS += -I$(INSTALL_ROOT)/$(CPUVARDIR)/$(PREFIX)/include
```

Comment:
I'm pretty sure this will drop the headers under `$(QNX_TARGET)/$(CPUVARDIR)/$(PREFIX)/include` when you're installing to a staging directory. It's a common issue for our ports.
We'll want:
```
-I$(QNX_TARGET)/$(CPUVARDIR)/$(PREFIX)/include \
-I$(INSTALL_ROOT)/$(CPUVARDIR)/$(PREFIX)/include
```

Reply:
I changed the file to match your fix above, but am still having trouble getting it to install in the staging area still- just to verify, I need to set both INSTALL_ROOT_nto to the staging area path and USE_INSTALL_ROOT to anything?

Reply:
Still not working in a69ecc2c454e20fe85e93a3518b308d6faa76c24 by setting `INSTALL_ROOT_nto` and `USE_INSTALL_ROOT`, but can force it to build in the staging area via `INSTALL_ROOT` (does not install in $QNX_TARGET in this case)

Reply:
Are you looking at the README for abseil-cpp?

Reply:
Yep - I didn't know there was a difference between setting them before vs after the make call (obtaining from environment vs command line). Fixed in 23fbade13f2e6db3602a2394de77d288dc9006d4
common.mk also missing a line that most of our other projects have, adding FLAGS to CFLAGS (`CFLAGS += $(FLAGS)`) - what is that doing? is it needed?

Reply:
Look at `$QNX_TARGET/usr/include/mk/qmacros.mk`, it's being set to some target env specific flags.